test/functional/ex_cmds/cd_spec.lua

```diff
@@ -0,0 +1,95 @@
+-- Specs for :cd, :tcd, :lcd
+
+local helpers = require('test.functional.helpers')
+local nvim, execute, eq, clear, eval, feed =
+  helpers.nvim, helpers.execute, helpers.eq, helpers.clear, helpers.eval, helpers.feed
+
+describe(':cd :tcd', function()
+  before_each(clear)
+
+  --it('sets to local directory for the program, tabs and windows', function()
+    local wd = function()
+      return eval('getcwd()')
+    end
+
+    -- Test both spellings: cd and chdir
+    -- Their variants differ in prefix (none, 't' and 'l')
+    for _, cmd in ipairs({'cd', 'chdir'}) do
+      describe('*' .. cmd, function()
+        it('works', function()
+          -- Store the initial working directory
+          local globalDir = wd()
+
+          -- Make tree of directories
+          local directories = {
+            {name = 'lua_test_dir_1', children =  {
```

Comment:
(By the way, I highly suggest using names which are unique to the test. Does not matter now, but I hope in the future test files will be run in different processes or threads; my actions though will not help if tests in the same file are run in parallel though.)

Specifically I use something like `Xtest-functional-ex_cmds-cd_spec.1`. This also helps to identify test responsible for some leftovers.